apps/server/src/app/base-fhir.controller.ts

```diff
@@ -284,6 +285,20 @@ export class BaseFhirController extends BaseController {
       const updateResults = await this.httpService.request(options).toPromise();
       const resource = updateResults.data;
 
+      const resources = (<ImplementationGuide> contextImplementationGuide).definition.resource;
+      const index = resources.findIndex(r => {
+        return r.reference.reference.indexOf(id) > 0;
+      });
+
+      if(index >= 0){
+        //If data.title exists, set to data.title. Else if data.name exists, set to data.name. Else if data.title and data.name don't exist, do nothing.
+        (<ImplementationGuide> contextImplementationGuide).definition.resource[index].name =
```

Comment:
We should also make sure that the resource has a title or name before updating the resource. Some resources don't have title/name at all... So we shouldn't bother in those cases. i.e.

```
if (data.title || data.name) {
  // set ig.definition.resource[index].name
  // PUT the resource
}
```

Reply:
Updated with requested conditional.